app/components/edit/EditServices.jsx

```diff
@@ -32,21 +31,16 @@ class EditServices extends Component {
    * @returns {void}
    */
   handleServiceChange(key, service) {
-    const { services } = this.state;
-    services[key] = service;
-    this.setState({
-      services,
-    }, () => {
-      this.props.handleServiceChange(this.state);
-    });
+    const { handleServiceChange } = this.props;
+    handleServiceChange({ services: { [key]: service } });
   }
 
   /* @method addService
    * @description Creates a brand new service
    */
   addService() {
-    const { existingServices } = this.state;
-    const newUUID = this.state.uuid - 1;
+    const { existingServices, uuid } = this.state;
+    const newUUID = uuid - 1;
```

Comment:
Wait a minute... I know this is existing code, but this is funky. We should probably just generate a new random UUID here.

Reply:
Oh wait, these aren't even UUIDs, it starts as -1 and then each time you add a service it subtracts 1 from it. I have no idea what is going on here or why. I guess we can just ignore this for now - wasn't really the point of this MR anyway.

Reply:
Yeah, I went through the exact same thought process here. I dug a little further and it's relying on the `-1` factor, since we have some _other_ code the partitions the services by ones with a positive "uuid" and ones with a negative "uuid". What it's really trying to do is keep track of new services which don't have an ID, so we really should handle that more explicitly.